2023/YoctoSummit-2023.11/fdo/fido-device-onboarding.md

```diff
@@ -0,0 +1,516 @@
+Onboarding Server will not accept the device as legitimate, leading to a
+failure in the provisioning and onboarding process.
+
+---
+
+# What is FDO?
+
+.center[
+<img src="/img/cert_chain_3.svg" height="460px">
+]
+
+???
+
+- This is how it looks in the end with all the protocols highlighted
+- What mattered for us most was the late binding provisioning
+- We needed to implement it inside of a Yocto project for a client
+- FIDO Alliance only specified the theoretical background -> implementations
+- All implementations have to implement those protocols in some way
+
+---
+
+# Existing implementations of FDO protocol
+
+.center[
+
+.right-column50[
+<img src="/img/lf_edge_logo.png" height="130px">
+]
+
+.left-column50[
+<img src="/img/securedeviceonboard_icon_color.png" height="130px">
+]
+
+### FDO Project
+
+]
+
+- [client-sdk-fidoiot](https://github.com/fido-device-onboard/client-sdk-fidoiot)
+- [pri-fidoiot](https://github.com/fido-device-onboard/pri-fidoiot)
+
+???
+
+- `client-sdk-fidoiot` is a direct descendant of [secure-device-onboard/client-sdk](https://github.com/secure-device-onboard/client-sdk)
+    + Written in C
+    + Currently supports only x86 Linux machines (Ubuntu, RHEL, Debian)
+    + Is planned to support Yocto at some point
+        - "The configurations mentioned in the document are not supported yet.
+        This document will be updated in a future release when the source code
+        implementation is updated to support the same."
+        - This line is not present in the `client-sdk` repo
+- The protocols can be implemented via the [pri-fidoiot](https://github.com/fido-device-onboard/pri-fidoiot)
+- Looks promising
+    + Has the potential to maybe suit the Yocto environment better
+- Hard pass for now
+
+---
+
+# Existing implementations of FDO protocol
+
+.center[
+<img src="/img/rustacean.svg" height="130px">
+
+### [fido-device-onboard-rs](https://github.com/fdo-rs/fido-device-onboard-rs)
+
+]
+
+.left-column50[
+- `fdo-client-linuxapp`
+- `fdo-rendezvous-server`
+- `fdo-owner-onboarding-server`
+- `fdo-serviceinfo-api-server`
+]
+
+.right-column50[
+- `fdo-manufacturing-client`
+- `fdo-manufacturing-server`
+- `fdo-owner-tool`
+]
+
+???
+
+- Developed by Fedora IoT as of recently, now the `fdo-rs` company
+    + Reached out to them on matrix
+    + The move to fdo-rs org was intended to represent that the FDO
+    implementation is really a community effort that is distro-agnostic.
+    + primary maintainers are Red Hatters working on Fedora IoT + RHEL for Edge
+- Modular - functionalities in form of crates
+- Looked most promising and faster to implement
+    + "How hard can it be to port rust crates to Yocto?"
+
+---
+
+.center[
+<img src="/img/rustacean-flat-sad.svg" height="145px">
+
+### "That should be easy right?"
+
+]
+
+???
+
+- Turns out cross-compiling rust is harder than it seemed at first
+
+---
+
+# State of rust in Yocto
```

Comment:
```suggestion
# State of Rust in Yocto
```

Reply:
[Addresed](https://github.com/3mdeb/conferences/blob/yocto_summit_2023_fdo/2023/YoctoSummit-2023.11/fdo/fido-device-onboarding.md?plain=1#L303) as well as in other instances